### PR TITLE
DOCTEAM-1225: adds nftables section

### PR DESCRIPTION
**Pull request by Amrita42**
### PR creator: Description

Describe the overall goals of this pull request.
Document nftables as the default backend for firewalld

### PR creator: Are there any relevant issues/feature requests?

[DOCTEAM-1225](https://jira.suse.com/browse/DOCTEAM-1225)

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [x] SLE 15 SP5/openSUSE Leap 15.5
  - [x] SLE 15 SP4/openSUSE Leap 15.4
  - [x] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2

- SLE 12
  - [ ] SLE 12 SP5

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by Amrita42** [comment]
[book-security_en.pdf](https://github.com/user-attachments/files/16156644/book-security_en.pdf)

**Review by jsegitz — CHANGES_REQUESTED** [review]
I added a couple of comments. Maybe add an additional example for a basic NAT config

File: xml/security_firewall.xml [hunk 1]
```diff
@@ -965,13 +965,35 @@ nfs-rpc
...
+       </para>
+      </listitem>
+      <listitem>
+       <para>
+        One framework for both the IPv4 and IPv6 protocols
+       </para>
+      </listitem>
+      <listitem>
+       <para>
+        Rules are applied automatically instead of fetching, updating and storing a complete rule set.
```
Comment (mgerstner) [inline xml/security_firewall.xml]: Do you mean `atomically` here? Automatically doesn't make much sense to me.

File: xml/security_firewall.xml [hunk 2]
```diff
@@ -965,13 +965,35 @@ nfs-rpc
...
+        </para>
+      </listitem>
+      <listitem>
+       <para>
+        Monitor trace events in the <emphasis>nft</emphasis> tool and debugging and tracing in the rule set <emphasis>nftrace</emphasis>.
```
Comment (mgerstner) [inline xml/security_firewall.xml]: The second part of the sentence doesn't look very sensical. How about:

`and debug and trace the rule set via <emphasis>nftrace</emphasis>`

File: xml/security_firewall.xml [hunk 3]
```diff
@@ -993,7 +993,14 @@ nfs-rpc
         Monitor trace events in the <emphasis>nft</emphasis> tool and debugging and tracing in the rule set <emphasis>nftrace</emphasis>.
         </para>
       </listitem>
-        </itemizedlist>
+      <listitem>
+       <para>
+      The <command>nft</command> command line tool complies the rule set into a VM bytecode in netlink format.
```
Comment (mgerstner) [inline xml/security_firewall.xml:993]: ~complies~ -> compiles

into VM bytecode

File: xml/security_firewall.xml [hunk 4]
```diff
@@ -993,7 +993,14 @@ nfs-rpc
...
+      During the rule set retrieval, the VM bytecode in netlink format is decompiled back to its original rule set representation.
+      nft acts as both a compiler and decompiler.
```
Comment (mgerstner) [inline xml/security_firewall.xml]: `decompiling` is not really a common term in this context. I would rather says something like `is translated back into`.

File: xml/security_firewall.xml [hunk 5]
```diff
@@ -961,7 +961,150 @@ nfs-rpc
...
+  <screen>&prompt.user; cat /etc/nftables.conf
+    #!/usr/sbin/nft -f
+
+    flush ruleset
+
+    # This matches IPv4 and IPv6
+    table inet filter {
+        # chain names are up to you.
+        # what part of the traffic they cover,
+        # depends on the type line.
```
[the remark this answers is not in the document]
Comment (mgerstner) [reply xml/security_firewall.xml:1012]: Maybe the `table` keyword was meant here.

File: xml/security_firewall.xml [hunk 6]
```diff
@@ -961,7 +961,150 @@ nfs-rpc
      <command>firewall-rpc-helper.py --help</command>.
     </para>
    </sect3>
-  </sect2>
+   <sect3 xml:id="sec-security-firewall-firewalld-nftables">
+    <title>nftables as the default backend </title>
+    <para>
+    The default backend for &firewalld; is nftables. nftables is a framework by the Linux <systemitem>netfilter</systemitem> project
+  that provides packet filtering, network address translation (NAT) and other packet mangling. nftables reuses the existing Netfilter
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/security_firewall.xml:968]: Yes, good point

File: xml/security_firewall.xml [hunk 7]
```diff
@@ -961,7 +961,150 @@ nfs-rpc
...
+    </para>
+    <itemizedlist mark="bullet" spacing="normal">
+        <listitem>
+       <para>
+        One framework for both the IPv4 and IPv6 protocols
+       </para>
+      </listitem>
+      <listitem>
+       <para>
+        Rules are applied automatically instead of fetching, updating and storing a complete rule set
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/security_firewall.xml:983]: correct atomically :)

File: xml/security_firewall.xml (hunk 1 above)
Comment (Amrita42) [reply xml/security_firewall.xml]: yes correct , its corrected

File: xml/security_firewall.xml (hunk 4 above)
Comment (Amrita42) [reply xml/security_firewall.xml:995]: Could you please elaborate more , I didnt understand

File: xml/security_firewall.xml (hunk 2 above)
Comment (Amrita42) [reply xml/security_firewall.xml]: Yes , this is much better , implemented :)

File: xml/security_firewall.xml (hunk 4 above)
Comment (mgerstner) [reply xml/security_firewall.xml:995]: I would phrase it `The VM bytecode in netlink format is translated back into the rule set representation.`

File: xml/security_firewall.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/security_firewall.xml:1012]: Yes correct , its the type table as per my understanding and re looking https://progress.opensuse.org/news/112

File: xml/security_firewall.xml (hunk 4 above)
Comment (Amrita42) [reply xml/security_firewall.xml:995]: ok implemented , also removed the sentence, nft acts as both a compiler and decompiler , as its already explained

File: xml/security_firewall.xml [hunk 8]
```diff
@@ -961,7 +961,150 @@ nfs-rpc
...
+  <listitem>
+   <para>
+   The <literal>base_checks</literal> chain handles all packets that are related to established connections
+   This ensures that incoming packets for outgoing connections are not blocked
+    </para>
+  </listitem>
+  <listitem>
+   <para>
+
+    ICMP and IGMP packets are allowed by utilizing  a set and type names
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/security_firewall.xml]: Good point

**Comment by Amrita42** [comment]
[book-security_en.pdf](https://github.com/user-attachments/files/16347454/book-security_en.pdf)

@jsegitz ptal , counters and another basic NAT config file has been added . Thanks

**Comment by jsegitz** [comment]
lgtm

**Review by jsegitz — APPROVED** [review]

**Comment by Amrita42** [comment]
@dariavladykina please peer review . thanks

File: xml/security_firewall.xml [hunk 9]
```diff
@@ -961,7 +961,174 @@ nfs-rpc
      <command>firewall-rpc-helper.py --help</command>.
     </para>
    </sect3>
-  </sect2>
+   <sect3 xml:id="sec-security-firewall-firewalld-nftables">
```
Comment (dariavladykina) [inline xml/security_firewall.xml:964]: I recommend collecting all nftables-related subsections under this subsection, i.e. this is `sect3`, and all the rest pertaining to nftabes (up to More information, incl.) should be `sect4`.

**Review by dariavladykina — COMMENTED** [review]
Hi Amrita, please see my suggestions here. Thanks!

File: xml/security_firewall.xml [hunk 10]
```diff
@@ -961,7 +961,174 @@ nfs-rpc
...
+  </listitem>
+  <listitem>
+   <para>
+   The <literal>base_checks</literal> chain handles all packets that are related to established connections
+   This ensures that incoming packets for outgoing connections are not blocked.
+    </para>
+  </listitem>
+  <listitem>
+   <para>
+    ICMP and IGMP packets are allowed by utilizing a set and type names.
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply xml/security_firewall.xml:1104]: I would prefer to retain the original sentence

File: xml/security_firewall.xml (hunk 9 above)
Comment (Amrita42) [reply xml/security_firewall.xml:964]: done

**Pull request merged**